lib/FeatureFlag.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace PostHog;
+
+const LONG_SCALE = 0xfffffffffffffff;
+
+class FeatureFlag
+{
+    public static function matchProperty($property, $propertyValues)
+    {
+        $key = $property["key"];
+        $operator = $property["operator"] ?? "exact";
+        $value = $property["value"];
+
+        if (!array_key_exists($key, $propertyValues)) {
+            throw new InconclusiveMatchException("Can't match properties without a given property value");
+        }
+
+        if ($operator == "is_not_set") {
+            throw new InconclusiveMatchException("can't match properties with operator is_not_set");
+        }
+
+        $overrideValue = $propertyValues[$key];
+
+        if ($operator == "exact") {
+            if (is_array($value)) {
+                return in_array($overrideValue, $value);
+            }
+            return $value == $overrideValue;
+        }
+
+        if ($operator == "is_not") {
+            if (is_array($value)) {
+                return !in_array($overrideValue, $value);
+            }
+            return $value !== $overrideValue;
+        }
+
+        if ($operator == "is_set") {
+            return array_key_exists($key, $propertyValues);
+        }
+
+        if ($operator == "icontains") {
+            return strpos(strtolower(strval($overrideValue)), strtolower(strval($value))) !== false;
+        }
+
+        if ($operator == "not_icontains") {
+            return strpos(strtolower(strval($overrideValue)), strtolower(strval($value))) == false;
+        }
+
+        if ($operator == "regex") {
+            return preg_match($value, $overrideValue);
```

Comment:
What happens when the regex isn't valid here? PHP noob so not sure 😋 . Should be in the tests though, will double check